### PR TITLE
ARC-170 Restyle the app

### PR DESCRIPTION
**Pull request by y0ungy3**
#beautiful

### CODE REVIEW

**Comment by CalicoCoder** [comment]
I think it looks pretty good all around here are two small things I found to fix.

- Use cursor: pointer for rows on main file list

- The search tab needs some padding to sides of the text, right now it is a little jumpy when you hover over the options for finding a file.

I also have two other things that are more opinion based changes so let me know what you think.

- Maybe change Archivist words in top left to our logo

- Margin under the nav bar makes thing look a little weird, the sidebar divider ends just before some random whitespace. Not sure exactly what to do but I tried quickly putting something together.

**Current**
current

**Option**
option

If you want that option you would just remove border radius and margin bottom from navbar. Add margin top to the home div.

**Comment by y0ungy3** [comment]
yeah I was trying to find where the random white space was. You should commit that change Rob, it looks good. 

For the logo on top left, I think it looks kinda weird or out of place since every other items on the nav bar is grey and the logo would just be really colorful. Maybe have the logo grey but when you hover over it, it turns colorful? I don't know lol

Also the hr divider in the settings screen, do you think it looks kinda weird? I was thinking of either putting everything into one big grey box (that's how it is right now) or separate each section into their own grey box and get rid of the divider.

I need to fix the input boxes in the options in the search tab, like the length

**Comment by CalicoCoder** [comment]
The dividers do look a little out of place but not too terrible. It doesn't look that bad without them entirely either.

That idea for the logo sounds good as well if you think you can make it fit the styling.

File: src/views/sidebar/components/sidebar.scss [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 
 .sidebar {
-    height: calc(100vh -  #{$action-bar-height});
+    height: 99%;
     width: 300px;
-    background-color: $white;
-    border-left: 1px solid $dark-blue;
+    background-color: $bg-primary-color;
+    border-left: 1px solid $secondary-color;
     overflow: scroll;
```
Comment (tjboswell) [inline src/views/sidebar/components/sidebar.scss]: can get rid of this, not needed

File: src/views/sidebar/components/sidebar.scss [hunk 2]
```diff
@@ -16,27 +16,27 @@
     &-tabs {
         display: flex;
         height: 25px;
-        border-color: $dark-blue;
+        border-color: $darker-secondary-color;
         border-style: solid;
         border-width: 0 0 1px;
```
Comment (tjboswell) [inline src/views/sidebar/components/sidebar.scss]: add a `width: 100%;` here

**Comment by tjboswell** [comment]
Looks good, there are some changes that need to be made though:

**action bar**

- The action bar is rounded, which looks off to me. At least the top corners should not be rounded
- The search icon should be to the right of search bar on the action bar
- Also, I like the idea of having the colored icon for the home link on the action bar, let me know if you want to do it/need me to make the icons

**main screen**

- The hover color for elements should be the primary (purple) color. For example, the page change arrows, sidebar toggle hover, and sidebar tab hover are all gray
- The no results message has a border around it which should be hidden. (do a search for some random tags)

**advanced search**

- The search 'cards' have weird looking inputs. Right now the dropdowns are the same color as the background which makes it hard to read.
- Hovering over a row's x changes the background instead of the icon. The buttons also aren’t aligned with the row
- Dropdowns for metadata should fit on same line, not 2
- Tags filter and/or dropdown is not aligned properly

**upload screen**

- I think that upload button should be below data entry. Seems weird if you have to scroll back up to upload a new item.
- Font/spacing seems off in general and the page takes up a lot of vertical space. I think you can reduce the spacing, font sizes, header padding, etc to make it a little more compact

**settings screen**

- The button to delete an item type is currently below metadata fields, I think it would make more sense as an X or “delete” next to each item type in the top list
- Change password shouldn’t be a button, should just display inputs on page load
- The logo isn't square and looks stretched out

Also, I know we are planning on getting rid of bootstrap, will you do that under this branch or separate?

**Review by tjboswell — CHANGES_REQUESTED** [review]

**Comment by mr2ert** [comment]
@y0ungy3 For your last commit, can you please make it a separate PR all on its own?  I know this is odious, but it'll really help with the documentation.

**Comment by mr2ert** [comment]
@tjboswell 

- "The button to delete an item type is currently below metadata fields, I think it would make more sense as an X or “delete” next to each item type in the top list"
  - IMHO this is fine as it is because this should be a very rare action.  It also aligns with the idea of that list being a selection of "what I want to see/work on".

**Comment by y0ungy3** [comment]
@mr2ert you want to keep my last commit in here for now til this PR is in tonight since it's a really small change, then all other bugs can branch off of master from now on or reverse the last commit on this branch...then branch off of master and do my last commit changes on a different branch?

also i made changes to others comments on here that I think should be change, the ones I didn't do cause i dont think theyre necessary and more of personal preferences.

**Comment by mr2ert** [comment]
@y0ungy3 "reverse the last commit on this branch...then branch off of master and do my last commit changes on a different branch?" This.

Could you indicate what you did/didn't change based on other's comments? (there are so many + so many lines of code/commits it is hard to tell)  Kind of like what I did above for one of Tyler's suggested changes.

**Comment by y0ungy3** [comment]
Rob's Comment:

X    Use cursor: pointer for rows on main file list

X    The search tab needs some padding to sides of the text, right now it is a little jumpy when you hover over the options for finding a file.
X    Margin under the nav bar makes thing look a little weird, the sidebar divider ends just before som random whitespace. Not sure exactly what to do but I tried quickly putting something together.

Tyler's Comment:
action bar

X    The action bar is rounded, which looks off to me. At least the top corners should not be rounded
X   The search icon should be to the right of search bar on the action bar
    Also, I like the idea of having the colored icon for the home link on the action bar, let me know if you want to do it/need me to make the icons

main screen

    The hover color for elements should be the primary (purple) color. For example, the page change arrows, sidebar toggle hover, and sidebar tab hover are all gray
 X   The no results message has a border around it which should be hidden. (do a search for some random tags)

advanced search

    The search 'cards' have weird looking inputs. Right now the dropdowns are the same color as the background which makes it hard to read.
X    Hovering over a row's x changes the background instead of the icon. The buttons also aren’t aligned with the row
X    Dropdowns for metadata should fit on same line, not 2
X    Tags filter and/or dropdown is not aligned properly

upload screen

X    I think that upload button should be below data entry. Seems weird if you have to scroll back up to upload a new item.
X    Font/spacing seems off in general and the page takes up a lot of vertical space. I think you can reduce the spacing, font sizes, header padding, etc to make it a little more compact

settings screen

    The button to delete an item type is currently below metadata fields, I think it would make more sense as an X or “delete” next to each item type in the top list
    Change password shouldn’t be a button, should just display inputs on page load
X    The logo isn't square and looks stretched out


x means been changed

**Comment by tjboswell** [comment]
First, we're still using bootstrap; I thought we had agreed to get rid of it. It's not worth removing it now because it would take too much time, but it's causing a lot of inconsistencies in our styles. For example:

- We have purple buttons on the sidebar and upload screen. However in the settings page, we have buttons with gray/white gradients.
- The checkboxes on the upload screen are styled differently and have an animation while the ones in the search sidebar tab are plain.
- The fonts for text inputs on the search tab/upload screen/settings screen are different and the text is larger. 
- (not just bootstrap, but related): We are aligning buttons different in different components. For example, the metadata tab left-aligns buttons but the search tab centers them. 

Those are the main issues that I see with it.

There are still a few issues that I think need to be fixed (doesn't have to be this PR).

- I still think the upload screen is way too large and spread out. Font-sizes, margins, and padding can be substantially reduced. I also think the right side can be expanded to be wider. Right now it seems very cramped. The upload icon does not need to take up 40% of the width
- There are some alignment issues. I know we created a ticket for the action bar, but there are also issues with the search filters like this: 
screen shot 2017-05-04 at 6 26 09 pm
- There are some color issues for active/hovered elements. These include the arrows to navigate pages, the tabs in the sidebar, and the edited metadata titles (especially on the darker gray which makes it unreadable)
screen shot 2017-05-04 at 6 33 12 pm

**Comment by y0ungy3** [comment]
@tjboswell all the changes to your previous comment should be there now. Take a look.

Only the action bar and the list for categories in the settings that I use bootstrap for cause it's a lot easier & faster to do. Otherwise, everything else are not bootstrap.

**Comment by tjboswell** [comment]
looks good to me now. there are still some issues but those will be handled by other tickets.

**Comment by mr2ert** [comment]
@tjboswell Can you accept the changes at the bottom of the PR?

**Review by tjboswell — APPROVED** [review]

**Pull request merged**